packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -117,18 +119,30 @@ export function UserListTable() {
   const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
   const [isDownloading, setIsDownloading] = useState(false);
 
-  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [filtersSearchState] = useFiltersSearchState();
+  const columnFilters = useMemo(() => {
+    return (filtersSearchState.activeFilters || [])
+      .map((filter) => ({
+        id: filter.f,
+        value: filter.v,
+      }))
+      .filter((filter) => {
+        // Keep the empty array state on the client side,
+        // but do not send it to the actual query.
+        if (Array.isArray(filter.value) && filter.value.length === 0) {
+          return false;
+        }
+        return true;
+      });
+  }, [filtersSearchState.activeFilters]);
```

Comment:
We already have `?activeFilters=` on the search state (URL), but kept `columnFilters` separately from `onColumnFilterChange` below. I think it's unnecessary to keep two sources, so I've made `columnFilters` out of `useFiltersSearchState()`.